### ISSUE TITLE
binder notebook runtime?

### ISSUE THREAD

**Issue by jGaboardi** [martinfleis/sdss22-workshop#3]
I just tried to run the notebook without changing anything from 2021 (Znojmo, Czechia). Here are the runtimes:
* local run:
  * ~ 5 minutes
  * macOS | 2.3 GHz Quad-Core Intel Core i7 | 32 GB 3733 MHz LPDDR4X
* binder run:
  * notebook processes for more than 30 minutes then crashes

**Comment by martinfleis** [comment]
Does the binder run through or gets stuck? We need to fix that.

**Comment by jGaboardi** [comment]
Getting stuck. But now also having trouble locally. May be a browser issue. Going to restart everything and try again.

**Comment by jGaboardi** [comment]
But the notebook *is* a memory hog. Was it for the previous tutorial?
<img width="625" alt="Screen Shot 2022-09-08 at 5 49 18 PM" src="https://user-images.githubusercontent.com/8590583/189232159-46e29245-b2f2-4a9c-aad4-620a4642c65d.png">

**Comment by martinfleis** [comment]
> But the notebook is a memory hog. Was it for the previous tutorial?

Not sure but the issue may be explore. That dumps the whole gdf to the browser memory to show it.

> Getting stuck

Do you know which step?

**Comment by jGaboardi** [comment]
> > But the notebook is a memory hog. Was it for the previous tutorial?
> 
> Not sure but the issue may be explore. That dumps the whole gdf to the browser memory to show it.
> 
> > Getting stuck
> 
> Do you know which step?

I can try to explore this further. I'll see about running via Binder once #6 is merged to figure out which cell(s) are the culprit(s).

**Comment by martinfleis** [comment]
For some reason, I am not able to plot in Binder. Or rather - it takes ages and I lose patience. The rest seem to run but takes quite a while, esp. clustergram.

**Comment by jGaboardi** [comment]
hmmm. Let's reopen then. Maybe St. Gallen is too bing then?

**Comment by martinfleis** [comment]
I don't think so, static plotting should be fine with that. Not sure what is going on there.

**Comment by martinfleis** [comment]
So, it works. Just takes an unbearable amount of time to create a plot.

**Comment by jGaboardi** [comment]
FWIW, I am playing with the binder now and not noticing a issue for plotting other that a ton of issues are being thrown [related to Python 3.7](https://github.com/martinfleis/sdss22-workshop/issues/10).

```
/srv/conda/envs/notebook/lib/python3.7/site-packages/geopandas/plotting.py:740: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  if mapclassify.__version__ < LooseVersion("2.4.0"):
/srv/conda/envs/notebook/lib/python3.7/site-packages/setuptools/_distutils/version.py:346: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  other = LooseVersion(other)
/srv/conda/envs/notebook/lib/python3.7/site-packages/geopandas/plotting.py:74: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  if mpl >= LooseVersion("3.4") or (mpl > LooseVersion("3.3.2") and "+" in mpl):
/srv/conda/envs/notebook/lib/python3.7/site-packages/setuptools/_distutils/version.py:346: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  other = LooseVersion(other)
```

**Comment by jGaboardi** [comment]
the Python 3.10 pin cleared up the "new" `DeprecationWarning`s that were thrown in the Python 3.7 environment. Now the time/memory hog only appears to be all the warnings the were fixed in pysal/momepy#386 and pysal/momepy#394. So once we have a fresh patch release, we can make sure that is the case (which is was when running locally).

**Comment by jGaboardi** [comment]
> the Python 3.10 pin cleared up the "new" `DeprecationWarning`s that were thrown in the Python 3.7 environment. Now the time/memory hog only appears to be all the warnings the were fixed in [pysal/momepy#386](https://github.com/pysal/momepy/issues/386) and [pysal/momepy#394](https://github.com/pysal/momepy/issues/394). So once we have a fresh patch release, we can make sure that is the case (which is was when running locally).

Yep, by adding in a warning catcher to cell 48 of the binder the cell (and remainder of the binder) runs smoothly and in decent time. So that `momepy` patch release should be just what the doctor ordered.

**Comment by jGaboardi** [comment]
All warning now handled properly.
